airflow/contrib/operators/databricks_operator.py

```diff
@@ -0,0 +1,196 @@
+    :type notebook_task: dict
+    :param new_cluster: Specs for a new cluster on which this task will be run.
+        *EITHER* ``new_cluster`` *OR* ``existing_cluster_id`` should be specified.
+        https://docs.databricks.com/api/latest/jobs.html#jobsclusterspecnewcluster
+    :type new_cluster: dict
+    :param existing_cluster_id: ID for existing cluster on which to run this task.
+        *EITHER* ``new_cluster`` *OR* ``existing_cluster_id`` should be specified.
+    :type existing_cluster_id: string
+    :param libraries: Libraries which this run will use.
+        https://docs.databricks.com/api/latest/libraries.html#managedlibrarieslibrary
+    :type libraries: list of dicts
+    :param run_name: The run name used for this task.
+        By default this will be set to the Airflow ``task_id``. This task_id is a
+        required parameter of the superclass ``BaseOperator``.
+    :type run_name: string
+    :param timeout_seconds: The timeout for this run. By default a value of 0 is used
+        which means to have no timeout.
+    :type timeout_seconds: int32
+    :param extra_api_parameters: Extra parameters which will be merged with parameters
+        listed above. This may be used if additional features are added to the
+        ``api/2.0/jobs/runs/submit`` endpoint.
+    :type extra_api_parameters: dict
+    :param databricks_conn_id: The name of the connection to use.
+        By default and in the common case this will be ``databricks_default``.
+    :type databricks_conn_id: string
+    """
+    # Databricks brand color (blue) under white text
+    ui_color = '#1CB1C2'
+    ui_fgcolor = '#fff'
+
+    def __init__(
+            self,
+            spark_jar_task=None,
+            notebook_task=None,
+            new_cluster=None,
+            existing_cluster_id=None,
+            libraries=None,
+            run_name=None,
+            timeout_seconds=0,
+            extra_api_parameters=None,
+            databricks_conn_id='databricks_default',
+            **kwargs):
+        """
+        Creates a new ``DatabricksSubmitRunOperator``.
+        """
+        super(DatabricksSubmitRunOperator, self).__init__(**kwargs)
+        self.spark_jar_task = spark_jar_task
+        self.notebook_task = notebook_task
+        self.new_cluster = new_cluster
+        self.existing_cluster_id = existing_cluster_id
+        self.libraries = libraries
+        self.run_name = kwargs['task_id'] if run_name is None else run_name
+        self.timeout_seconds = timeout_seconds
+        if extra_api_parameters is None:
+            self.extra_api_parameters = {}
+        else:
+            self.extra_api_parameters = extra_api_parameters
+        self.databricks_conn_id = databricks_conn_id
+        self._validate_parameters()
+
+    def _validate_parameters(self):
+        """
+        Validates the parameters provided to this operator.
+        """
+        if not self._validate_oneof(self.spark_jar_task, self.notebook_task):
```

Comment:
It would be simpler to let server handle those validations. One use case is that in the future if we introduce new task types, users cannot use `extra_api_parameters` to specify the new task type.